Documentation/Yarn-Spec.md

```diff
@@ -0,0 +1,636 @@
+
+#### if
+
+The _if_ command is the opening command of flow control and is broken up into two parts, the keyword and the expression and must be in that order.
+The _keyword_ is the text `if`.
+The _expression_ is an expression.
+The expression must resolve to a boolean.
+
+`<<if $boldness > 1>>` is an example of an if command, `<<if 1>>` is an example of an invalid if, it is invalid because the expression does not resolve to a boolean.
+
+#### elseif
+
+The _elseif_ command is an optional component of flow control and allows for additional flow to be expressed.
+The command works in a fashion very similar to the if command.
+The command is broken up into two parts, the keyword and the expression and must be presented in that order.
+The _keyword_ is the text `elseif`.
+The _expression_ is an expression.
+The expression must resolve to a boolean.
+
+The minimum mumber of required elseif commands must be zero.
+The maximum number of allowed elseif commands is unspecified but should be greater than zero.
+An elseif command must not exist without an if command and must go after the if command.
+
+#### else
+
+The _else_ command is an optional component of flow control and allows for additional flow to be expressed.
+The command only has a single component, the keyword `else`.
+
+There must only be a single else command (if any) per flow control.
+The else command must go after the if and any elseif commands.
+The else must not exist without an if command.
+
+The example of the else command is `<<else>>`.
+
+#### endif
+
+The _endif_ command is the final element of flow control and is comprised solely of the keyword `endif`.
+The endif must be present whenever there is flow control and must go after the if and any elseif or else commands.
+The endif exists to allow the implementing program know when the scope of the other elements in the flow control has ended.
+`<<endif>>` is the example of the endif command.
+
+#### Scope and Blocks
+
+For the flow control to be useful there needs to be yarn statements which are run only when their appropriate expression evaluates to true.
+Flow control allows for blocks of statements to be scoped to their commands.
+A _block_ is a collection of statements that are scoped to a particular part of the flow control.
+
+The _scope_ of a block is determined by the flow control commands and associates each block with a command.
+The if, elseif, and else commands all have a block associated with them.
+The block of statements for a command start from the first statement after the command up until the next command in the flow control.
+When dealing with nested flow control the deepest set of flow control commands are to be the ones that can first assume another command closes their scope.
+
+While it is common for writers to indent their blocks relative to their scope it must not be used by the implementing program to determine scope.
```

Comment:
Recommend "to determine scope when handling flow control statements." as whitespace is syntactically significant for options